mantidproject/mantid
# Add peak controls to the workbench fitting tool

**Description of work.**
Added peak controls to the fitting tool: vertical lines that can be moved with the mouse marking the peak's centre and its width. Added the guess curve plotting.

**To test:**
1. Open a 1D plot and the fit browser.
2. Right click on the plot and select `Add Peak` from the menu.
3. Click inside the plot. 3 vertical lines must show up. And a peak function added to the browser.
4. Try to move the lines with the mouse and see that peak parameters change.
5. Select a different peak function from the context menu. Check that the added peak has the selected type.

Fixes #24434

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
